src/computer.cpp

```diff
@@ -369,7 +369,7 @@ void computer::activate_function(computer_action action, char ch)
                                     break;
                                 }
                                 if (!found_item) {
-                                    g->m.add_item_or_charges(x1, y1, sewage);
+                                    g->m.add_item_or_charges( tripoint( x1, y1, g->m.get_abs_sub().z ), sewage );
```

Comment:
This conversion from 2D to 3D should definitively remain in `map`. This ensures all those conversion use the same logic for the z-component. This also makes it much easier to see which 2D overloads are still used and which aren't.

A simple check for example is to remove the definition of the overload, but keep the declaration in the class and recompile - the linker will complain when the overload is still used.

And this code is less efficient: it has another function call (`map::get_abs_sub`) instead of accessing the member directly from within the 2D overload.